### PR TITLE
Refactor overlay to allow components to not necessarily be notice driven

### PR DESCRIPTION
**Pull request by allisonking**
Half of https://github.com/ethyca/fides/issues/3809

### Description Of Changes

Refactor of `fides-js` components so they can handle something other than privacy notices. See [this message for more context](https://ethyca.slack.com/archives/C05E74TR0BB/p1690407064504709)

Mostly this involves breaking existing components down so that their shared parts are their own component, but notice specific logic can be pulled into a different component. Everything should still work as it did before, but now if it receives an experience with `component=tcf_overlay`, it'll render the `<TcfTabs/>` instead!

### Code Changes

* [x] Add TCF types based on https://github.com/ethyca/fides/pull/3804
* [x] Add cypress scaffolding so we can use a mock payload for in development TCF work
* [x] Refactor `Overlay` into `NoticeOverlay` and `TcfOverlay`
* [x] Refactor consent buttons and notice toggles to separate out privacy notice logic
* [x] Render TcfTabs (and remove their notice dependency as well)

### Steps to Confirm

* Everything works the same
* Use cypress to see the TCF version

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/privacy-center/cypress/e2e/consent-banner.cy.ts [hunk 1]
```diff
@@ -2,141 +2,16 @@ import {
...
-  ConsentMechanism,
-  EnforcementLevel,
-  FidesOptions,
-  PrivacyExperience,
-  UserConsentPreference,
-  UserGeolocation,
-} from "fides-js/src/lib/consent-types";
-import { FidesEndpointPaths } from "fides-js/src/services/fides/api";
-
-enum OVERRIDE {
```
Comment (allisonking) [inline clients/privacy-center/cypress/e2e/consent-banner.cy.ts:21]: moved this over to `stubs.ts` so that my new cypress test could share these handy functions

File: clients/privacy-center/cypress/e2e/consent-banner-tcf.cy.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    cy.fixture("consent/experience_tcf.json").then((experience) => {
+      stubConfig({
+        options: {
+          isOverlayEnabled: true,
+        },
+        experience: experience.items[0],
+      });
+    });
+  });
+  it("can render the TCF modal", () => {});
```
Comment (allisonking) [inline clients/privacy-center/cypress/e2e/consent-banner-tcf.cy.ts:16]: doesn't test for anything right now, but this is a convenient way to "see" the tcf modal

File: clients/fides-js/src/lib/consent.tsx [hunk 3]
```diff
@@ -39,10 +41,12 @@ export const initOverlay = async ({
         document.body.prepend(parentElem);
       }
 
+      // DEFER: use a separate TCF bundle instead of this if statement
```
Comment (allisonking) [inline clients/fides-js/src/lib/consent.tsx:44]: okay SO the main idea here is that there is a different entrypoint component for TCF vs Notices. Rather than just our handy `Overlay` component, we now have `NoticeOverlay` and `TcfOverlay`, both of which wrap the original `Overlay` component. this is handy for code splitting, because (once we move away from the if/else logic, and instead have `fides-tcf.ts` render `TcfOverlay`) rollup should be able to figure out which components we need for each bundle

File: clients/fides-js/src/components/index.ts [hunk 4]
```diff
@@ -1,2 +1 @@
-export { default as NoticeToggles } from "./NoticeToggles";
```
Comment (allisonking) [inline clients/fides-js/src/components/index.ts:1]: this was from my experiments of trying to use the toggles in privacy center, and isn't used

File: clients/fides-js/src/components/Tcf/TcfOverlay.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import { TcfConsentButtons } from "../ConsentButtons";
+import { OverlayProps } from "../types";
+import TcfTabs from "./TcfTabs";
+
+const TcfOverlay: FunctionComponent<OverlayProps> = ({
+  experience,
+  options,
+  cookie,
+}) => {
+  // TODO: how will we initialize TCF data?
```
Comment (allisonking) [inline clients/fides-js/src/components/Tcf/TcfOverlay.tsx:18]: all of these TCF specific TODOs I figured should go in a separate PR since this refactor is already large

File: clients/fides-js/src/components/Overlay.tsx [hunk 6]
```diff
@@ -1,58 +1,51 @@
-import { h, FunctionComponent } from "preact";
```
Comment (allisonking) [inline clients/fides-js/src/components/Overlay.tsx:1]: Overlay has gone through a big refactor to remove everything that made it PrivacyNotice dependent. Instead, `NoticeOverlay` controls all notice related logic, while `Overlay` is mostly just in charge of controlling showing banner/modal logic

File: clients/fides-js/src/components/DataUseToggle.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
+import { ComponentChildren, VNode, h } from "preact";
+import { useDisclosure } from "../lib/hooks";
+import Toggle from "./Toggle";
+
+interface DataUse {
+  key: string;
+  name?: string;
+}
+
+const DataUseToggle = ({
```
Comment (allisonking) [inline clients/fides-js/src/components/DataUseToggle.tsx:10]: changed `NoticeToggle` to `DataUseToggle` to make it less specific to notices. It takes an obj with a `key` and `name` instead of a notice object so that the TCF models can in theory use it as well

File: clients/fides-js/src/components/notices/NoticeToggles.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import { h } from "preact";
+
+import { ConsentMechanism, PrivacyNotice } from "../../lib/consent-types";
+
+import Divider from "../Divider";
+
+import { GpcBadgeForNotice } from "../GpcBadge";
+import DataUseToggle from "../DataUseToggle";
+
+const NoticeToggles = ({
```
Comment (allisonking) [inline clients/fides-js/src/components/notices/NoticeToggles.tsx:10]: Like `NoticeOverlay`, this is a notice specific version of a list of an existing component (`NoticeToggle` --> `DataUseToggles`)

File: clients/fides-js/src/components/ConsentButtons.tsx [hunk 9]
```diff
@@ -5,29 +5,78 @@ import {
   PrivacyExperience,
   ConsentMechanism,
   PrivacyNotice,
+  ExperienceConfig,
 } from "../lib/consent-types";
 
-type NoticeKeys = Array<PrivacyNotice["notice_key"]>;
-
 const ConsentButtons = ({
```
Comment (allisonking) [inline clients/fides-js/src/components/ConsentButtons.tsx:11]: there are now 3 components in this file:

* `ConsentButtons`
* `NoticeConsentButtons`
* `TcfConsentButtons`

The original `ConsentButtons` had a lot of notice specific logic in it, so a refactoring as needed!

**Review by eastandwestwind — APPROVED** [review]
This is a big one, nice work @allisonking !

Also a note that I've regression tested this PR
